libs/utils/decorate.js

```diff
@@ -1,29 +1,88 @@
+import { createTag, getConfig, } from '../utils/utils.js';
 
 export function decorateButtons(el, size) {
   const buttons = el.querySelectorAll('em a, strong a');
   if (buttons.length === 0) return;
-  buttons.forEach((button) => {
+  buttons.forEach((button, idx) => {
     const parent = button.parentElement;
-    const buttonType = parent.nodeName === 'STRONG' ? 'blue' : 'outline';
+    let buttonType = parent.nodeName === 'STRONG' ? 'blue' : 'outline';
+    if (buttonType === 'outline' && button.firstChild.nodeName === 'STRONG') buttonType = 'fill';
     button.classList.add('con-button', buttonType);
     if (size) button.classList.add(size); /* button-L, button-XL */
-    parent.insertAdjacentElement('afterend', button);
-    parent.remove();
+    const actionArea = button.closest('p');
+    actionArea?.classList.add('action-area');
+    if (idx === buttons.length - 1) {
+      button.closest('p')?.nextElementSibling?.classList.add('supplemental-text', 'body-XL');
+    }
   });
-  const actionArea = buttons[0].closest('p');
-  actionArea.classList.add('action-area');
-  actionArea.nextElementSibling?.classList.add('supplemental-text', 'body-XL');
 }
 
-export function decorateIcons(el) {
+export function decorateIconArea(el) {
   const icons = el.querySelectorAll('.icon');
   icons.forEach((icon) => {
     icon.parentElement.classList.add('icon-area');
     if (icon.textContent.includes('persona')) icon.parentElement.classList.add('persona-area');
   });
 }
 
+export async function getSVGsfromFile(path, selectors) {
+  if (!path) return null;
+  const resp = await fetch(path);
+  if (!resp.ok) return null;
+
+  const text = await resp.text();
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(text, 'image/svg+xml');
+
+  if (!selectors) {
+    const svg = doc.querySelector('svg');
+    if (svg) return [{ svg }];
+    /* c8 ignore next */
+    return null;
+  }
+
+  return selectors.map((selector) => {
+    const symbol = doc.querySelector(`#${selector}`);
+    const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    while (symbol.firstChild) svg.appendChild(symbol.firstChild);
+    [...symbol.attributes].forEach((attr) => svg.attributes.setNamedItem(attr.cloneNode()));
+    svg.classList.add('icon-milo', `icon-milo-${selector}`);
+    svg.removeAttribute('id');
+    return { svg, name: selector };
+  });
+}
+
+function getIconList(el) {
+  const icons = el.querySelectorAll('span.icon');
+  let iconList = [];
+  icons?.forEach(async (i) => {
+    const iconName = i.classList[1].replace('icon-milo-', '');
+    if (!iconName) return;
+    if (!iconList.includes(iconName)) iconList.push(iconName);
+  });
+  return iconList;
+}
+
+export async function decorateIconsInBlock(el) {
+  const { miloLibs, codeRoot } = getConfig();
+  const base = miloLibs || codeRoot;
+  const icons = el.querySelectorAll('span.icon');
+  const iconList = getIconList(el) || [];
```

Comment:
Seems unnecessary considering getIconList will always return an array
```suggestion
  const iconList = getIconList(el);
```